final-project/Figures/api-component.mmd

```diff
@@ -0,0 +1,29 @@
+C4Container
+title Component diagram for APIProxy for OpenTogetherTube
+
+    Person(customer, Customer, "A user of OTT", $tags="v1.0")
+
+    Container_Boundary(c1, "OpenTogetherTube") {
+    Container(browser, "Browser", "Web Browser", "Used by the user to access the application", "v1.0")
+    Container(server, "Server", "Typescript", "Serves as the backend for the OpenTogetherTube platform", "v1.0")
+    Container(redis, "Redis", "In-memory data structure store", "Stores session information for authenticated users and used for some caching", "v1.0")
+    Container(postgres, "Postgres", "Object-relational database", "v1.0")
+    Container(reverseproxy, "HTTP API Reverse Proxy", "NGINX", "Distributes incoming traffic among monoliths", "v1.0")
```

Comment:
This is not correct. As we discussed previously, the balancer needs to route certain room-specific requests to the correct monolith and this is **not** possible with nginx. Please do not include nginx in any diagrams.

The reverse proxy is something that the balancer needs to implement. Please let me know if we need to set up time to clarify this.

Reply:
sorry, my oversight, i can take out the nginx but other than that is the diagram right? how else do i include the reverse proxy if not how I have it?

Reply:
The term "Reverse Proxy" is more of a concise summary of the logic of receiving an HTTP request, and making an equivalent request to another server, waiting for a response, and returning that response to the original requester. This is not the correct diagram to describe this behavior.

The diagram we have here describes OTT's typical deployment today, without the load balancer. This is information we have already diagrammed. If you want to keep this, that's fine, but its definitely not "Component diagram for APIProxy for OpenTogetherTube".

If you're trying to do a component diagram for the _balancer_, then you need to describe the different modules _inside the balancer_. That would be api proxy, and the websocket proxy.

Reply:
for now im more concerned with how we need a component diagram. let me know what to label it and i can put it in the slides for the presentation

Reply:
For the presentation, we can use any the diagrams in #51 so we can say we have a component diagram. For this PR, this diagram will either need to change, or be removed.